bsp/stm32f10x-HAL/drivers/drv_sdcard.c

```diff
@@ -0,0 +1,330 @@
+/*
+ * File      : drv_sdcard.c
+ * This file is part of RT-Thread RTOS
+ * COPYRIGHT (C) 2018, RT-Thread Development Team
+ *
+ * The license and distribution terms for this file may be
+ * found in the file LICENSE in this distribution or at
+ * http://www.rt-thread.org/license/LICENSE
+ *
+ * Change Logs:
+ * Date           Author         Notes
+ * 2018-01-13     Liu2guang      the first version.
+ */
+
+#include <rtthread.h>  
+#include <rtdevice.h> 
+#include "drv_sdcard.h"
+#include "stm32f1xx.h" 
+
+#define SD_TIMEOUT ((uint32_t)0x100000)
+ 
+static SD_HandleTypeDef hsdcard = 
+{
+    .Instance                 = SDIO,
```

Comment:
Please not use 'c99' feature of C programming language. Otherwise, you should add 'c99' to the 'CFLAGS'.